mmdeploy/codebase/mmagic/deploy/video_super_resolution.py

```diff
@@ -0,0 +1,244 @@
+        elif self.model_cfg.get("test_pipeline", None):
+            test_pipeline = self.model_cfg.test_pipeline
+        else:
+            test_pipeline = self.model_cfg.val_pipeline
+
+        # check if the input is a video
+        file_extension = osp.splitext(video)[1]
+        if file_extension in VIDEO_EXTENSIONS:
+            video_reader = mmcv.VideoReader(video)
+            # load the images
+            data = dict(img=[], img_path=None, key=video)
+            for frame in video_reader:
+                data["img"].append(np.flip(frame, axis=2))
+
+            # remove the data loading pipeline
+            tmp_pipeline = []
+            for pipeline in test_pipeline:
+                if pipeline["type"] not in [
+                    "GenerateSegmentIndices",
+                    "LoadImageFromFile",
+                ]:
+                    tmp_pipeline.append(pipeline)
+            test_pipeline = tmp_pipeline
+        else:
+            # the first element in the pipeline must be
+            # 'GenerateSegmentIndices'
+            if test_pipeline[0]["type"] != "GenerateSegmentIndices":
+                raise TypeError(
+                    "The first element in the pipeline must be "
+                    f'"GenerateSegmentIndices", but got '
+                    f'"{test_pipeline[0]["type"]}".'
+                )
+
+            # specify start_idx and filename_tmpl
+            test_pipeline[0]["start_idx"] = self.extra_parameters["start_idx"]
+            test_pipeline[0]["filename_tmpl"] = self.extra_parameters["filename_tmpl"]
+
+            # prepare data
+            sequence_length = len(glob.glob(osp.join(video, "*")))
+            lq_folder = osp.dirname(video)
+            key = osp.basename(video)
+            data = dict(
+                img_path=lq_folder, gt_path="", key=key, sequence_length=sequence_length
+            )
+
+        # compose the pipeline
+        test_pipeline = Compose(test_pipeline)
+        data = test_pipeline(data)
+        results = data["inputs"].unsqueeze(0) / 255.0  # in cpu
+        data["inputs"] = results
+        return data
+
+    def create_input(
+        self,
+        video: InputsType,
+        input_shape: Sequence[int] = None,
+        data_preprocessor: Optional[BaseDataPreprocessor] = None,
+    ) -> Tuple[Dict, torch.Tensor]:
+        """Create input for editing processor.
+
+        Args:
+            imgs (str | np.ndarray): Input image(s).
+            input_shape (Sequence[int] | None): A list of two integer in
+             (width, height) format specifying input shape. Defaults to `None`.
+            data_preprocessor (BaseDataPreprocessor): The data preprocessor
+                of the model. Default to `None`.
+
+        Returns:
+            tuple: (data, img), meta information for the input image and input.
+        """
+        data = self.preprocess(video)
+        return data, BaseTask.get_tensor_from_input(data)
+
+    def forward(
```

Comment:
this task class should not have forward function.

Reply:
Thanks. I will change based on your recommendatation

Reply:
I have added the end2endmodel in video_super_resolution_model.py. and move the foward function there. Unfortunately, I have met same problem